Jack/Seed.fs

```diff
@@ -32,87 +58,69 @@ module Seed =
         let s = s0 &&& Int32.MaxValue
 
         // TODO remove crashUnless
-
-        // The integer variables s1 and s2 must be initialized to values
-        // in the range [1, 2147483562] and [1, 2147483398] respectively. [1]
         crashUnless (s >= 0) "s >= 0"
-        let q, s1 = Math.DivRem(s, 2147483562)
-        crashUnless (q >= 0) "q >= 0"
-        let s2 = q % 2147483398
-    
-        Seed (s1 + 1, s2 + 1)
+
+        { Value = int64 s
+          Gamma = goldenGamma }
+
+    let private mix64 (s0 : int64) : int64 =
+        let s = s0
+        let s = (s ^^^ (s >>> 33)) * 0xff51afd7ed558ccdL
+        let s = (s ^^^ (s >>> 33)) * 0xc4ceb9fe1a85ec53L
+        s ^^^ (s >>> 33)
+
+    let private mix64variant13 (s0 : int64) : int64 =
+        let s = s0
+        let s = (s ^^^ (s >>> 30)) * 0xbf58476d1ce4e5b9L
+        let s = (s ^^^ (s >>> 27)) * 0x94d049bb133111ebL
+        s ^^^ (s >>> 31)
+
+    let private bitCount (s0 : int64) : int =
+        let s = s0 - ((s0 >>> 1) &&& 0x5555555555555555L)
+        let s = (s &&& 0x3333333333333333L) + ((s >>> 2) &&& 0x3333333333333333L)
+        let s = (s + (s >>> 4)) &&& 0x0f0f0f0f0f0f0f0fL
+        let s = s + (s >>> 8)
+        let s = s + (s >>> 16)
+        let s = s + (s >>> 32)
+        (int s) &&& 0x7f
```

Comment:
It's a shame there is no .NET intrinsic for `POPCNT`, this should be a single instruction with SSE4